### ISSUE TITLE
🤖 [AUTO] Implement Strategy: fama_french_next_day_predictive

### ISSUE THREAD

**Issue by kpakpooluwaseyi** [kpakpooluwaseyi/Agentic-Algo-Strategies#441]
## 🤖 Auto-Generated Strategy Request

**Source:** Machine Trading by Ernest P. Chan.pdf (strategy 2)
**Extracted:** 2025-12-25T02:43:51.543642

---

1.  **Strategy Name**: fama_french_next_day_predictive
2.  **Strategy Type**: Factor-based, Predictive
3.  **Timeframe**: Daily (positions are entered based on next-day prediction, implying daily rebalancing)
4.  **Instruments**: Stocks (specifically, S&P 500 Index stocks)

5.  **Entry Rules**:
    -   **Prediction Model**: Use a multivariate, multiple linear regression model to predict the next-day excess return (stock return minus risk-free rate) for each stock. The model is trained on historical Fama-French factors and historical next-day stock returns.
    -   **Long Entry**: At the current day's market close, identify the top 50 stocks with the largest *predicted* next-day returns. Enter a long position in these stocks for the next trading day.
    -   **Short Entry**: At the current day's market close, identify the bottom 50 stocks with the smallest (or most negative) *predicted* next-day returns. Enter a short position in these stocks for the next trading day.

6.  **Exit Rules**:
    -   **Time-based**: Positions are implied to be re-evaluated and potentially exited/re-entered daily, given the "next-day return" prediction. Specific take-profit or stop-loss levels are not mentioned.

7.  **Indicators Required**:
    -   **Fama-French Factors**:
        -   Excess Market Return (Mkt-RF): Market return minus the risk-free rate.
        -   HML (High Minus Low): Returns of a basket of high book-to-market stocks minus that of a basket of low book-to-market stocks.
        -   SMB (Small Minus Big): Returns of a basket of small-market-capitalization stocks minus that of a basket of big-market-capitalization stocks.
    -   **Price Data**: Mid-quotes at market close for S&P 500 stocks.
    -   **Risk-free Rate**: Needed to calculate excess returns.

8.  **Risk Management**:
    -   **Position Sizing**: Long 50 stocks and short 50 stocks. Implicitly assumes equal capital allocation per stock for simplicity (e.g., if one were to "buy $1 of the stock").
    -   **Maximum Risk per Trade**: Not specified.
    -   **Risk-Reward Ratio**: Not specified.

9.  **Additional Notes**:
    -   **Historical Data**: CRSP survivorship-bias-free database (Jan 3, 2007, to Dec 31, 2013) for stock prices. Fama-French factor data from Professor French's website.
    -   **Data Processing**: Use mid-price (half of bid and ask) at market close to avoid effects of widened bid-ask spreads.
    -   **Model Training**: The first half of the historical data is used as a training set to estimate factor loadings for each stock separately.
    -   **Performance Note**: In the provided example, the strategy showed very strong in-sample performance (CAGR 242%, Sharpe 3.7) but generated negative returns out-of-sample, suggesting the Fama-French factors were "not terribly useful for short-term predictions" in this direct application, indicating potential overfitting or lack of true predictive power for next-day returns.

---

## Implementation Instructions

Please implement this strategy using the `backtesting.py` framework with the following requirements:

1. **File Location:** `strategies/fama_french_next_day_predictive.py`
2. **Data Path:** Use `data/BTC-USD-15m.csv` for backtesting
3. **Indicators:** Use `talib` or `pandas_ta` (NOT backtesting.py's built-in indicators)
4. **Template:** Follow existing strategies in the `strategies/` folder

### Required Components:
- Strategy class inheriting from `Strategy`
- Proper indicator initialization using `self.I()`
- Entry/exit logic in `next()` method
- Default parameters as class attributes
- Risk management (stop loss, take profit)

### Data Format:
```
datetime, open, high, low, close, volume
2023-01-01 00:00:00, 16531.83, 16532.69, 16509.11, 16510.82, 231.05
```

### Output:
- Run backtest with default parameters
- Print stats and create HTML plot
- Save results to `results/temp_result.json`
